misc-scripts/xref_mapping/XrefParser/BaseParser.pm

```diff
@@ -566,42 +565,34 @@ sub upload_xref_object_graphs {
        #######################################################################
        # if there are dependent xrefs, add xrefs and dependent xrefs for them
        #######################################################################
+     DEPENDENT_XREF:
        foreach my $depref (@{$xref->{DEPENDENT_XREFS}}) {
 	 my %dep = %{$depref};
 
-	 #################
-	 # Insert the xref
-	 #################
+	 #####################################
+	 # Insert the xref and get its xref_id
+	 #####################################
 	 # print "inserting $dep{ACCESSION},$dep{VERSION},$dep{LABEL},$dep{DESCRIPTION},$dep{SOURCE_ID},${\$xref->{SPECIES_ID}}\n";
 	 $xref_sth->execute($dep{ACCESSION},
-			   $dep{VERSION} || 0,
-			   $dep{LABEL} || $dep{ACCESSION},
-			   $dep{DESCRIPTION} || '',
+			   $dep{VERSION} // 0,
+			   $dep{LABEL} // $dep{ACCESSION},
+			   $dep{DESCRIPTION},
 			   $dep{SOURCE_ID},
 			   $xref->{SPECIES_ID},
                            'DEPENDENT');
 
-	 #####################################
-	 # find the xref_id for dependent xref
-	 #####################################
-	 $xref_id_sth->execute(
-                   $dep{ACCESSION},
-                   $dep{SOURCE_ID},
-                   $xref->{SPECIES_ID} );
-         my $dep_xref_id = ($xref_id_sth->fetchrow_array())[0];
-
-	 if(!(defined $dep_xref_id) || $dep_xref_id ==0 ){
-	   print STDERR "acc = $dep{ACCESSION} \nlink = $dep{LINKAGE_SOURCE_ID} \n".$dbi->err."\n";
-	   print STDERR "source = $dep{SOURCE_ID}\n";
+         my $dep_xref_id = $xref_sth->{'mysql_insertid'};
```

Comment:
As long as this remains a DBI statement handle, perhaps you could use DBI's last_insert_id() instead of the driver method? I'd like for us to become less backend specific, as the core API is.

Reply:
I've used the _mysql_insertid_ way because that's what _add_xref()_ uses - but I am all for being database-agnostic.